### ISSUE TITLE
DatePicker (Calendar) TODOs

### ISSUE THREAD

**Issue by CRImier** [ZeroPhone/ZPUI#101]
A simple calendar would be easy to make in ZPUI. However, if you think about it, we could really use a calendar-like UI element for picking dates - similar to how there's a `PathPicker` element, and there's an `Utils=>File browser` app that's essentially a `PathPicker` and a few more lines of code. Desired functions:

- [x] Traditional view - month/year on top in a large font, 5 lines for weeks
- [x] Accepting a start value (`__init__` argument) - good if it expects same keyword arguments as it returns in the dict
  - [x] Fallback value - today's day/month/year
- [x] Either accepting a return value or executing a callback (like `PathPicker` does)
- [x] Being able to set date output format as per `strftime` standards (for either the return value or the callback argument), but still returning a dict by default
- [ ] If you're on first week and press UP, it should go to previous month (same weekday) - same for reverse.

Optional nice-to-haves:
- [x] Show weeks either starting from Sunday instead of Monday (`__init__` argument)
- [x] Showing week numbers (`__init__` argument)
- [ ] Add a grid of years (on some special key) - to quicker pick from the years (i.e. you need to select your date-of-birth)

**Comment by CRImier** [comment]
Thanks to @derivmug for the work - now there are only TODOs left =)

**Comment by piajesse** [comment]
when you press the `KEY_ENTER` button it should open up a list of events on said day, and if there are none you should open a create new event menu, or something of the likes

**Comment by CRImier** [comment]
Yep, that'll be possible once there's a callback addition, I'll work on Calendar app plugins (and implement the event tracking functionality through plugins) after that's added

**Comment by piajesse** [comment]
maybe make it so that when you press `KEY_F1` it goes smaller on the calendar size, like we could have Day, Week, month, and maybe a year view, and `KEY_F2` would go bigger? e.g. Day > Week

**Comment by CRImier** [comment]
Most of the items are done - thanks to @derivmug !
